src/node.cc

```diff
@@ -3687,6 +3683,15 @@ void Init(int* argc,
     }
   }
 
+#ifdef __POSIX__
+  // Block SIGPROF signals when sleeping in epoll_wait/kevent/etc.  Avoids the
+  // performance penalty of frequent EINTR wakeups when the profiler is running.
+  // Only for v8.log profiling. CpuProfiler API expects to have evenly distributed samples.
```

Comment:
Just a comment, no action needed: the reason it's done unconditionally is that `--prof` is not the only way to do sample-based profiling (besides the `v8::CpuProfiler` API.)  I can live with the change though.

You can drop the `#ifdef __POSIX__`.  It fails with UV_ENOSYS on Windows but that's okay because the return value isn't checked.  Maybe you can add a comment to that effect.